### PR TITLE
Update QDK to .NET6.0 (qdk-python)

### PR DESCRIPTION
**Pull request by ricardo-espinoza**
We're migrating the QDK to the most recent Long Time Support version of the .NET framework. For details about this change, refer to the original issue [Q#C:1224](https://github.com/microsoft/qsharp-compiler/issues/1224).

As part of this change, we're:

- Re-targeting all .NetCoreApp3.1 binaries to .NET6.0
- Updating Docker images, samples and templates.
- Libraries using .NetStandard2.1 are not affected by this change.
- The minimum supported .NET version in the QDK will also be updated from 3.1 to 6.0

This is a multi-repo change, and all changes will need to be merged concurrently.

### CODE REVIEW

**Review by guenp — APPROVED** [review]

**Comment by ltalirz** [comment]
Just wondering - is the updated QDK version already out or is this notebook now broken?

I believe I'm running the latest version of the qsharp conda package and, as of today, the QDK docs [still mention to install .NET 3.1](https://docs.microsoft.com/en-us/azure/quantum/install-command-line-qdk#prerequisite).

Wouldn't it have made sense to wait until the new QDK release until merging this PR?

**Comment by ricardo-espinoza** [comment]
@ltalirz , you are correct. This PR was merged prematurely by mistake. We'll roll it back and merge again so it goes simultaneously with the other 8 parts.

**Comment by ricardo-espinoza** [comment]
PR #208 was opened to revert this.

**Pull request merged**
